service/controller/network/v1/resource/ipam/create.go

```diff
@@ -1,9 +1,56 @@
 package ipam
 
-import "context"
+import (
+	"context"
+	"fmt"
+	"net"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/giantswarm/cluster-operator/service/controller/network/v1/key"
+)
 
 // EnsureCreated takes care of cluster subnet allocation when
 // ClusterNetworkConfig object is created.
 func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
+	clusterNetworkCfg, err := key.ToCustomObject(obj)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	clusterID := key.ClusterID(clusterNetworkCfg)
+
+	r.logger.LogCtx(ctx, "level", "debug", "message", "allocating subnet", "clusterID", clusterID)
+
+	if clusterNetworkCfg.Status.IP != "" {
+		// Subnet allocated. No need to do anything.
+		r.logger.LogCtx(ctx, "level", "debug", "message", "no need to allocate subnet", "clusterID", clusterID)
+		return nil
+	}
+
+	maskBits := key.ClusterNetworkMaskBits(clusterNetworkCfg)
+
+	mask := net.CIDRMask(maskBits, 32)
+
+	subnet, err := r.ipam.CreateSubnet(ctx, mask, clusterID)
+	if err != nil {
+		return microerror.Maskf(err, "subnet allocation failed with mask %s, clusterID: %s", net.IP(mask).String(), clusterID)
+	}
+
+	clusterNetworkCfg.Status.IP = subnet.IP.String()
+	clusterNetworkCfg.Status.Mask = net.IP(subnet.Mask).String()
+
+	_, err = r.g8sClient.CoreV1alpha1().ClusterNetworkConfigs(clusterNetworkCfg.GetNamespace()).UpdateStatus(&clusterNetworkCfg)
```

Comment:
Do you mean status of `{AWS,Azure,KVM}ClusterConfig`? With this, do you suggest dropping network controller and just implementing a resource that is used by provider specific controllers in `cluster-operator`?

Reply:
Yes. We do have shared resources right now, e.g. chartconfig. This could be one of them as well. It's kind of intuitive to see in the status of AWSConfing what range was allocated.

Reply:
> It's kind of intuitive to see in the status of AWSConfing what range was allocated

Just for verification, you mean `AWSClusterConfig`? I'm asking this because in general the controller managing CRD should manage `Status` of it, so for `cluster-operator` it would be `AWSClusterConfig`.

This would make sense and the overall code would very similar (`AWSConfig` resource must check some CR's status anyway) as with this implementation except without additional controller.

Reply:
Yes, you are right it should be AWSClusterConfig. Only the operator consuming the CR should manipulate the status.